sympy/ntheory/residue_ntheory.py

```diff
@@ -786,6 +798,16 @@ def _nthroot_mod_composite(a, n, m):
         a.append(list(y))
     return sorted(set(crt(m, list(i))[0] for i in cartes(*a)))
 
+def _nthroot_mod_composite(a, n, m):
+    """
+    Find the solutions to ``x**n = a mod m`` when m is not prime.
+    """
+    return _help(m,
+        lambda p: nthroot_mod(a, n, p, True) or [],
```

Comment:
when `all_roots` is True a list will be returned so this `or []` is not necessary.